samples/drivers/stepper/generic/src/main.c

```diff
@@ -0,0 +1,112 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 Jilay Sandeep Pandya.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/stepper.h>
+#include <zephyr/input/input.h>
+#include <zephyr/kernel.h>
+
+static const struct device *stepper = DEVICE_DT_GET(DT_ALIAS(stepper));
+
+enum stepper_mode {
+	STEPPER_MODE_PING_PONG_RELATIVE,
+	STEPPER_MODE_PING_PONG_ABSOLUTE,
+	STEPPER_MODE_ROTATE_CW,
+	STEPPER_MODE_ROTATE_CCW,
+};
+
+static atomic_t stepper_mode = ATOMIC_INIT(STEPPER_MODE_PING_PONG_RELATIVE);
+
+static int32_t ping_pong_target_position =
+	CONFIG_STEPS_PER_REV * CONFIG_PING_PONG_N_REV * DT_PROP(DT_ALIAS(stepper), micro_step_res);
+
+static K_SEM_DEFINE(stepper_generic_sem, 0, 1);
+
+static void stepper_callback(const struct device *dev, const enum stepper_event event,
+			     void *user_data)
+{
+	switch (event) {
+	case STEPPER_EVENT_STEPS_COMPLETED:
+		k_sem_give(&stepper_generic_sem);
+		break;
+	default:
+		break;
+	}
+}
+
+static void button_pressed(struct input_event *event, void *user_data)
+{
+	ARG_UNUSED(user_data);
+
+	if (event->value == 0) {
```

Comment:
I suggest checking also for `evt->type == INPUT_EV_KEY`, then if there's some other weird thing on the board it gets skipped